tasty-query/shared/src/main/scala/tastyquery/reader/tasties/TreeUnpickler.scala

```diff
@@ -296,6 +297,30 @@ private[tasties] class TreeUnpickler(
       case tpe                => throw TastyFormatException(s"unexpected type for readWithin: $tpe")
   end readWithin
 
+  private def readAnnotationsInModifiers(sym: Symbol, end: Addr)(using LocalContext): Unit =
```

Comment:
any reason not to read the annotations in `readModifiers`?

Reply:
Yes, I tried that first. It didn't work because we read the modifiers while creating all the Symbols. To read annotations, we need to read Trees, and we can only do that after all the Symbols of the file have been created.